compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -2440,6 +2439,31 @@ impl<'tcx> ConstantKind<'tcx> {
     }
 }
 
+/// An unevaluated (potentially generic) constant used in MIR.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord, TyEncodable, TyDecodable, Lift)]
+#[derive(Hash, HashStable)]
+pub struct Unevaluated<'tcx> {
+    pub def: ty::WithOptConstParam<DefId>,
+    pub substs: SubstsRef<'tcx>,
+    pub promoted: Option<Promoted>,
+}
+
+impl<'tcx> Unevaluated<'tcx> {
```

Comment:
Calling it just 'unevaluated' is very confusing. `mir::Unevaluated` doesn't show any relationship to constants. It should be `UnevaluatedConst` or so IMO.

Reply:
:shrug: don't have a strong opinion here, `ConstDefId` seems wrong, would be fine with `UnevaluatedConst`